### DISCUSSION TITLE
Gas meter with a sensor that has 2 reed contacts

### DISCUSSION THREAD

**Discussion by blissi** [arendst/Tasmota#16696] [Q&A] [answered]
Hi,

I would like to use Tasmota as a gas consumption meter. I have bought a sensor that consists of two reed contacts:
One is the counter that is normally open, and one is the control reed that is normally closed.

I'm no expert, but I believe the purpose of the control reed is to make reading the counter reliable (debounce issue that you have with only one reed contact). So, when the counter reed is closed, the other one is open and now you know that you are only allowed to increase the pulse once. When the control reed is then closed, you know that it is now allowed to increase the pulse again the next time the counter reed is closed. What do you think, am I right in this regard?

Is it possible to use such a sensor in Tasmota? I have set up the two GPIOs where I connected the reeds as counters and I can see the counters increase, but now I'm stuck.

The data sheet of the sensor is here (page 3 is the relevant page):
https://cdn.shopify.com/s/files/1/0435/3399/2089/files/metrix_impulzusado.pdf?v=1597743852

Thanks,
Blissi

**Comment by sfromis** [comment, marked as answer]
The PDF document states:
> The control contact reed detects failure situation like:
> * trials of blocking the counting circuit with a strong magnet
> * breaking cable connection between a gas meter and counting device

To me, this sounds simply like having two reed switches in different distances, in case one does not work. My expectation would be that them counting in unison would indicate a good setup, otherwise to consider if it could/should be improved.

Your expectation looks like a more complex scheme about the two reed switches not being independent, what is your basis for that?

**Reply by rollercontainer** [reply]
As I understood it, the second line is a sabotage contact to detect if the client is "optimizing" his consumption...

**Comment by blissi** [comment]
Oh ok, now I see :-)
So I will implement it now as the many tutorials in the web show.

Thank you!
